File: atv6/exerc6.go
package main
import "fmt"

type Motor struct {
    Tipo string
}

type Carro struct {
    Marca string
    Motor Motor
}

func (c Carro) detalhes() {
    fmt.Println("Carro:", c.Marca, "com motor", c.Motor.Tipo)
}

func main() {
    motor := Motor{"V8"}
    carro := Carro{"Ford", motor}
    carro.detalhes()
}

File: atv6/exerc6.py
class Motor:
    def __init__(self, tipo):
        self.tipo = tipo

class Carro:
    def __init__(self, marca, motor):
        self.marca = marca
        self.motor = motor

    def detalhes(self):
        print(f"Carro: {self.marca} com motor {self.motor.tipo}")

motor = Motor("V8")
carro = Carro("Ford", motor)
carro.detalhes()
